File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.py
class Solution:
    def smallestPalindrome(self, s: str) -> str:
        first_half = "".join(sorted(s[:len(s)//2]))
        return first_half + (s[len(s)//2] if len(s) % 2 == 1 else "") + first_half[::-1]
        

File: 3517-smallest-palindromic-rearrangement-i/3517-smallest-palindromic-rearrangement-i.cpp
class Solution {
public:
    string smallestPalindrome(string s) {
        vector<int> freq(26, 0);

        for (char c: s.substr(0, s.length()/2)) {
            freq[c-'a']++;
        }

        std::string result;
        for (int i=0; i<freq.size(); i++) {
            int count = freq.at(i);
            if (count > 0) result += string(count, char(static_cast<int>('a')+i));
        }
        std::string reversed_result(result.rbegin(), result.rend());
        return result + (s.length() % 2 == 1 ? string(1, s[s.length()/2]) : "") + reversed_result;
    }
};